store/src/java/com/zimbra/cs/service/mail/ToXML.java

```diff
@@ -1687,6 +1706,39 @@ private static Element encodeMessageAsMPHelper(boolean bestEffort, Element paren
         }
     }
 
+    private static List<Integer> findWord(String textString, String word) {
+        List<Integer> indexes = new ArrayList<Integer>();
+        String lowerCaseTextString = textString.toLowerCase();
+        String lowerCaseWord = word.toLowerCase();
+
+        int index = 0;
+        while(index != -1) {
+            index = lowerCaseTextString.indexOf(lowerCaseWord, index);
+            if (index != -1) {
+                indexes.add(index);
+                index++;
+            }
+        }
+        return indexes;
+    }
+
+    private static List<String> getSenderReceiverEmailAddresses(Element parent) {
```

Comment:
We would use the existing way to find the mail sender and recipients using the Message object.

Reply:
Refactored the code to get senders and recipients mail from message object